tools/c7n_azure/c7n_azure/resources/cost_management_export.py

```diff
@@ -0,0 +1,163 @@
+from c7n_azure.provider import resources
+from c7n_azure.query import QueryResourceManager
+from c7n_azure.utils import ThreadHelper
+
+from c7n.filters import Filter
+from c7n.utils import get_annotation_prefix as gap
+from c7n.utils import type_schema
+import logging
+
+import datetime
+
+
+log = logging.getLogger('c7n.azure.cost-management-export')
+
+
+@resources.register('cost-management-export')
+class CostManagementExport(QueryResourceManager):
+    """ Cost Management Exports for current subscription (doesn't include Resource Group scopes)
+
+    :example:
+
+    Returns all cost exports for current subscription scope
+
+    .. code-block:: yaml
+
+        policies:
+          - name: get-cost-management-exports
+            resource: azure.cost-management-export
+
+    """
+
+    class resource_type(QueryResourceManager.resource_type):
+        doc_groups = ['Cost']
+
+        service = 'azure.mgmt.costmanagement'
+        client = 'CostManagementClient'
+        enum_spec = ('exports', 'list', None)
+        default_report_fields = (
+            'name',
+            'location',
+            'resourceGroup',
+        )
+        resource_type = 'Microsoft.Compute/images'
+
+        @classmethod
+        def extra_args(cls, resource_manager):
+            scope = '/subscriptions/{0}'\
+                .format(resource_manager.get_session().get_subscription_id())
+            return {'scope': scope}
+
+
+@CostManagementExport.filter_registry.register('last-execution')
+class CostManagementExportFilterLastExecution(Filter):
+    """ Find Cost Management Exports with last execution more than X days ago.
+
+    :example:
+
+    Returns all cost exports that didn't run in last 30 days.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: find-stale-management-exports
+            resource: azure.cost-management-export
+            filters:
+              - type: last-execution
+                age: 30
+    """
+
+    schema = type_schema(
+        'last-execution',
+        required=['age'],
+        **{
+            'age': {'type': 'integer'}
+        }
+    )
+
+    def process(self, resources, event=None):
+        self.client = self.manager.get_client()
+        self.scope = 'subscriptions/{0}'.format(self.manager.get_session().get_subscription_id())
+        self.min_date = datetime.datetime.now() - datetime.timedelta(days=self.data['age'])
+
+        result, _ = ThreadHelper.execute_in_parallel(
+            resources=resources,
+            event=event,
+            execution_method=self._check_resources,
+            executor_factory=self.executor_factory,
+            log=log
+        )
+
+        return result
+
+    def _check_resources(self, resources, event):
+        result = []
+
+        for r in resources:
+            if gap('last-execution') in r:
+                continue
+            history = self.client.exports.get_execution_history(self.scope, r['name'])

```

Comment:
TBH, I am not familiar with this API. However, is it worth caching the result of `get_execution_history`? I would if it's large or expensive.

Reply:
Don't think so, at least not for this filter. We don't really need entire history, only the last item